packages/storage-driver-gcs/src/index.ts

```diff
@@ -88,6 +112,55 @@ export class DriverGCS implements Driver {
 			query = nextQuery;
 		}
 	}
+
+	get tusExtensions() {
+		return ['creation', 'termination', 'expiration'];
+	}
+
+	async createChunkedUpload(filepath: string, context: ChunkedUploadContext) {
+		const file = this.file(this.fullPath(filepath));
+
+		const [uri] = await file.createResumableUpload();
+
+		context.metadata!['uri'] = uri;
+
+		return context;
+	}
+
+	async writeChunk(filepath: string, content: Readable, offset: number, context: ChunkedUploadContext) {
+		const file = this.file(this.fullPath(filepath));
+
+		const stream = file.createWriteStream({
+			chunkSize: this.preferredChunkSize,
+			uri: context.metadata!['uri'] as string,
+			offset,
+			isPartialUpload: true,
+			resumeCRC32C: context.metadata!['hash'] as string,
+			metadata: {
+				contentLength: context.size || 0,
+			},
+		});
+
+		stream.on('crc32c', (hash: string) => {
+			context.metadata!['hash'] = hash;
+		});
+
+		let bytesUploaded = offset || 0;
+
+		content.on('data', (chunk: Buffer) => {
+			bytesUploaded += chunk.length;
+		});
+
+		await pipeline(content, stream);
+
+		return bytesUploaded;
+	}
+
+	async finishChunkedUpload(_filepath: string, _context: ChunkedUploadContext) {}
+
+	async deleteChunkedUpload(filepath: string, _context: ChunkedUploadContext) {
+		await this.delete(filepath);
+	}
```

Comment:
I think we should be deleting the chunked upload here 🤔 

```suggestion
	async deleteChunkedUpload(filepath: string, context: ChunkedUploadContext) {
		await this.deleteChunkedUpload(filepath, context);
	}
```

Reply:
The proposed change would be a recursive call to this function which is not desired. The current call is correct.

Reply:
You are very right! I had mistaken it for a GCS function 🫣 Thanks!!